### PR TITLE
Merging CZ19/Pixel into master

### PR DESCRIPTION
**Pull request by tjclement**
Making a draft PR to look into needed changes

### CODE REVIEW

File: firmware/components/micropython/lib/utils/pyexec.c
Comment (tjclement) [inline firmware/components/micropython/lib/utils/pyexec.c:504]: This fixes the python console not doing a newline when you press enter, but instead only doing a carriage return

File: firmware/main/main.c
Comment (tjclement) [inline firmware/main/main.c:39]: This is not mergeable as-is

File: firmware/components/driver_framebuffer/driver_framebuffer.c
Comment (tjclement) [inline firmware/components/driver_framebuffer/driver_framebuffer.c]: Not too sure about why this was changed to be honest

**Comment by tjclement** [comment]
Huge PR, but the relevant parts for a merge are not too bad. It's intended to get CZ19/Pixel merged back into master, and share the changes made for it. I think for the PR we have to focus on only 3 things (because the others are just python_module changes that don't change anything for other badges):

- sndmixer: this PR includes the changes done for CZ20. It includes some new features like looping and streaming wav and mp3 files, and changes the way it mixes multiple channels (namely, additive mixing instead of normalized mixing. The former is how most audio equipment works)

- framebuffer: some small changes, one of which I don't remember or understand anymore. See the comment I added for it. The other changes are to get it to skip lengthy compilation if it's not enabled in KConfig.

- main.c: we made some changed in startup and ota graphics that are not directly mergeable because they break other badges. Let's look at how this can be made neatly.

@renzenicolai what do you think?

**Review by renzenicolai — DISMISSED** [review]
Too much to check, if you could test that at least the SHA2017 badge still works with this change merged then I trust that this is an improvement :smile:

File: firmware/main/main.c
Comment (renzenicolai) [reply firmware/main/main.c:39]: Let's merge this anyway as this step is only used in our factory and not when people flash on their own (even when their flash has been wiped).

In fact: let's switch to a micropython based solution for this instead on new badges.

File: firmware/python_modules/pixel/mpu6050.py
Comment (renzenicolai) [inline firmware/python_modules/pixel/mpu6050.py:1]: We hebben nu ook een C driver voor deze chip, want die zat op MCH2021 prototype 1 :smile: 
Misschien leuk om daar ooit naar toe te wisselen.

**Review by renzenicolai — DISMISSED** [review]

**Comment by renzenicolai** [comment]
LGTM, strangely enough I think this whole PR can be merged without breaking any of the other badges.

There is a merge conflict that needs to be solved, but after that feel free to hit that merge button :smile:

**Review by renzenicolai — APPROVED** [review]

File: firmware/python_modules/pixel/mpu6050.py
Comment (tjclement) [reply firmware/python_modules/pixel/mpu6050.py:1]: Super nice, although specifically for CZ19 we need a modified driver (such as this python one) that has built-in retries and error checks, because of a hardware design issue (too weak pull-up values)

**Pull request merged**
